### ISSUE TITLE
Enumeratee package etc

### ISSUE THREAD

**Issue by thoughtpolice** [thoughtpolice/salt#7]

**Comment by thoughtpolice** [comment]
Considering `pipes-3.0` was just released, I think this package would fit perfectly here.

**Comment by tel** [comment]
Is there any idea how the "Networking" side of NaCl would fit in here? I don't know if the NaCl team has released ideas about how they want to solve networking.

**Comment by tel** [comment]
And yes! I think Gabriel's work on `pipes` really makes it shine as a best contender magic-fabric-pet-tunnel library, especially paired with other simplified libraries and APIs.

**Comment by thoughtpolice** [comment]
This is a pretty open design space but I think something reasonable can be done. This is why I recommended pipes, because it directly models bidirectionality you'll often see in communications. Sounds like a great use case for `encryptNM/decryptNM` :)

Unfortunately I'm punting this off a bit since we'll want some infrastructure surrounding `pipes-3.0` first (like networking or parsing infrastructure.) #6 will track the eventual network adapter, which will be low level. I think this is the proper level of abstraction. The basic idea:
- (`pipes` + `pipes-networking`) ++ (`salt` + `salt-networking`) = `salt-pipes`

So basically we should be able to tie the -network packages together.
